File: python/test/test_2709.py
from python.src.code_2709 import Solution
from python.test.python_deps.boilerplate import BoilerTest


def input_fetcher(case):
    return case["input"]["nums"]


def test():
    test = BoilerTest(2709, input_fetcher)
    solution = Solution().canTraverseAllPairs
    test.run_tests(solution)

File: cpp/test/test_2709.cpp
#include "../src/code_2709.cpp"
#include "cpp_deps/boilerplate.hpp"

using namespace std;

void test(Solution& sol, const json& input, const json& output) {
    vector<int> nums = input["nums"].get<vector<int>>();
    bool expected = output.get<bool>();
    bool result = sol.canTraverseAllPairs(nums);
    CHECK_EQ(result, expected);
}

TEST_CASE("") {
    TEST(2709);
}